code/modules/fishing/fish/chasm_detritus.dm

```diff
@@ -40,30 +40,28 @@ GLOBAL_LIST_INIT_TYPED(chasm_detritus_types, /datum/chasm_detritus, init_chasm_d
 		),
 	)
 
-/datum/chasm_detritus/proc/dispense_reward(turf/fishing_spot, turf/fisher_turf)
-	if (prob(default_contents_chance))
-		var/default_spawn = pick(default_contents[default_contents_key])
-		return new default_spawn(fisher_turf)
-	return find_chasm_contents(fishing_spot, fisher_turf)
+/datum/chasm_detritus/proc/dispense_reward(reward_path, mob/fisherman, turf/fishing_spot)
```

Comment:
This arg is inconsequential, unused and, because the reward path is the same as the type of this datum, redundant.

To be clear, that proc you're showing me in that snippet of code is actually
`/datum/fish_source/proc/dispense_reward()`, while this is `/datum/chasm_detritus/proc/dispense_reward()`. Same name, different procpath.

Reply:
sure, i'm in for a different name, to make it less puzzling.